include/natalie/backtrace.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "natalie/forward.hpp"
+#include "natalie/gc.hpp"
+#include "tm/hashmap.hpp"
+
+namespace Natalie {
+class Backtrace : public Cell {
+public:
+    struct Item {
+        const ManagedString *source_location;
+        const char *file;
+        size_t line;
+    };
+
+    void add_item(Env *env, const char *file, size_t line) {
+        // NATFIXME: Ideally we could store the env and delay building the code location name
```

Comment:
I tried doing exactly this and I notices some really weird behavior with it:

The Env object seems to be fine, but the Method object that is references on the Env ends up being 0x10 which will crash the program. I suspect the GC or something like this breaks it?

Reply:
Well that's not cool. :cry:

Reply:
Some of our Envs are stack-allocated and short-lived. Others are heap-allocated and get cleaned up by the GC. You can make a copy of a single Env, but that wouldn't fix the issue of other Envs in the call stack getting overwritten/cleaned up. I think this is the best we can do.